shared/bls/herumi/signature_test.go

```diff
@@ -191,3 +190,12 @@ func TestSignatureFromBytes(t *testing.T) {
 		})
 	}
 }
+
+func TestCopy(t *testing.T) {
+	signatureA := &Signature{s: bls12.HashAndMapToSignature([]byte("foo"))}
+	signatureB, ok := signatureA.Copy().(*Signature)
+	require.Equal(t, true, ok)
```

Comment:
to test it better you need to do `signatureA.s.Add(random_sig)` and then make sure sigA and sigB are unequal.